chia/wallet/wallet_node.py

```diff
@@ -120,7 +121,7 @@ def __init__(
         self.proof_hashes: List = []
         self.state_changed_callback = None
         self.wallet_state_manager = None
-        self.new_state_lock = None
+        self.new_state_lock = asyncio.Lock()
```

Comment:
This needs to happen in _start not in init, init can happen outside the asyncio running loop

Reply:
Thanks for pointing that out.  Why does it matter though?

Reply:
if you construct the object but you are not in an asyncio loop, you will get an error when you try to use it, something related to being in a different asyncio loop.

Reply:
Before I only tested that I could construct it outside an `async def`.  Now I've tried acquiring the lock and that seems ok, including running the code below in Python 3.7 locally.  The last test run looked ok outside of some installation issues, so we'll see what it looks like.  Hopefully they exercise this.  But sure, this is a dual purpose PR...  yuck.  I can extract this if it's going to hold things up.

https://replit.com/@altendky/UtterGiantTexts-1
```python
import asyncio


lock = asyncio.Lock()


async def main():
    async with lock:
        print("in the lock")


asyncio.run(main())
```
```
in the lock
```

Reply:
This caused issues in the past, so i am confused why it's working well here..

Reply:
Just a simplification, no need to push it in now.  I'll move the other bits forward without this.